Change the default cpufreq governor from ondemand to performance on Intel
hosts.

It is unlikely there will be a power consumption regression because Intel CPU
also support c-states for saving power which are still enabled.

AMD processors don't fully implement c-states so we have to leave cpu
frequency scaling enabled on these processors.

diff --git a/xen/arch/x86/cpu/intel.c b/xen/arch/x86/cpu/intel.c
index 863edac4cd7d..9a785eb624dc 100644
--- a/xen/arch/x86/cpu/intel.c
+++ b/xen/arch/x86/cpu/intel.c
@@ -12,6 +12,8 @@
 #include <asm/msr.h>
 #include <asm/trampoline.h>
 
+#include <acpi/cpufreq/cpufreq.h>
+
 #include "cpu.h"
 
 /*
@@ -687,6 +689,10 @@ static void cf_check init_intel(struct cpuinfo_x86 *c)
 
 	if (c == &boot_cpu_data && cpu_has(c, X86_FEATURE_ERMS))
 		setup_force_cpu_cap(X86_FEATURE_XEN_REP_MOVSB);
+
+	/* Default to performance cpufreq governor */
+	if (!cpufreq_opt_governor)
+		cpufreq_opt_governor = &cpufreq_gov_performance;
 }
 
 const struct cpu_dev __initconst_cf_clobber intel_cpu_dev = {
